backend/games/views/scores.py

```diff
@@ -0,0 +1,59 @@
+import django_filters
+
+from rest_framework.permissions import IsAuthenticated, AllowAny, IsAdminUser
+from rest_framework import pagination, mixins, viewsets
+from authentication.models.user import User
+from games.serializers import (
+    ScoreSerializer,
+    CreateScoreSerializer,
+)
+from games.models import Game, Score
+
+
+class ActionBasedPermission(AllowAny):
+    """
+    Grant or deny access to a view, based on a mapping in view.action_permissions
+    """
+
+    def has_permission(self, request, view):
+        for klass, actions in getattr(view, "action_permissions", {}).items():
+            if view.action in actions:
+                return klass().has_permission(request, view)
+        return False
```

Comment:
Could you explain what this does? It seems very generic so I'm surprised we have to reimplement it here

Reply:
It allows a viewset to have a specific permission class for each allowed action. Here I needed to have the permission class IsAdminUser for the CREATE action, and the permission class IsAuthenticated for the LIST action. (I found this solution on stackoverflow, didn't find better...)